client/src/main/java/io/split/client/SplitFactoryImpl.java

```diff
@@ -626,22 +628,43 @@ private static SSLContext buildSSLContext(SplitClientConfig config) throws IOExc
 
     private static HttpClientBuilder setupProxy(HttpClientBuilder httpClientbuilder, SplitClientConfig config) {
         _log.info("Initializing Split SDK with proxy settings");
-        DefaultProxyRoutePlanner routePlanner = new DefaultProxyRoutePlanner(config.proxy());
+        HttpHost proxyHost;
+        if (config.proxyConfiguration() != null && config.proxyConfiguration().getHost() != null) {
+            proxyHost = config.proxyConfiguration().getHost();
+        } else {
+            proxyHost = config.proxy();
+        }
+        DefaultProxyRoutePlanner routePlanner = new DefaultProxyRoutePlanner(proxyHost);
         httpClientbuilder.setRoutePlanner(routePlanner);
 
-        if (config.proxyUsername() != null && config.proxyPassword() != null) {
+        if ((config.proxyUsername() != null && config.proxyPassword() != null) ||
+                (config.proxyConfiguration() != null && config.proxyConfiguration().getProxyCredentialsProvider() != null &&
+                        config.proxyConfiguration().getProxyCredentialsProvider() instanceof io.split.client.dtos.BasicCredentialsProvider)) {
             _log.debug("Proxy setup using credentials");
+            String userName;
+            String password;
+            if (config.proxyUsername() != null && config.proxyPassword() != null) {
```

Comment:
Check that if both the old options and new Proxy configuration options are provided, the old options are ignored and a warning should be logged. Maybe something like:

`"[WARN] `proxyHost`, `proxyPort`, `proxyUsername` and `proxyPassword` configuration methods are deprecated. Please use the `ProxyConfiguration` object instead."`

Reply:
will do